src/set_theory/game/domineering.lean

```diff
@@ -66,58 +93,139 @@ open domineering_aux
 
 instance : has_well_founded (finset (ℤ × ℤ)) := ⟨measure finset.card, measure_wf finset.card⟩
 
-/-- We construct a domineering game from any finite subset of `ℤ × ℤ`. -/
+/-
+# Implementation note:
+It would be nice to just give the definition:
+```
 def domineering : finset (ℤ × ℤ) → pgame
 | b := pgame.mk
     (left b) (right b)
     (λ m, have _, from move_left_smaller b m,  domineering (move_left b m))
     (λ m, have _, from move_right_smaller b m, domineering (move_right b m))
+```
+And indeed that works, but I couldn't make it computational -- that is, I couldn't produce an
+instance of `short (domineering b)` that would reduce during `dec_trivial`.
+It appears that the reason for this is that this definition uses well-founded recursion,
+and we can't definitionally unfold --- only propositionally.
+
+Instead, we give the following auxiliary definition, which puts the fact that the number of
+board squares decreases at each step front and center.
+-/
+
+/-- An auxiliary definition for the construction of domineering games. -/
+def domineering' : Π (n : ℕ) (b : finset (ℤ × ℤ)), b.card = n → pgame
+| 0 b _ := pgame.mk (left b) (right b) (λ m, 0) (λ m, 0)
+| 1 b _ := pgame.mk (left b) (right b) (λ m, 0) (λ m, 0)
+| (n + 2) b h :=
+  pgame.mk (left b) (right b)
+    (λ m, domineering' n (move_left b m) begin have t := (move_left_card b m), rw h at t, simp at t, exact t, end )
+    (λ m, domineering' n (move_right b m) begin have t := (move_right_card b m), rw h at t, simp at t, exact t, end)
 
-@[simp] lemma domineering_left_moves (b : finset (ℤ × ℤ)) :
-  (domineering b).left_moves = left b :=
-by { rcases b with ⟨⟨b, _⟩|⟨h, t⟩, n⟩; refl }
-@[simp] lemma domineering_right_moves (b : finset (ℤ × ℤ)) :
-  (domineering b).right_moves = right b :=
-by { rcases b with ⟨⟨b, _⟩|⟨h, t⟩, n⟩; refl }
+/-- We construct a domineering game from any finite subset of `ℤ × ℤ`. -/
+def domineering (b : finset (ℤ × ℤ)) : pgame :=
+domineering' b.card b rfl
+
+-- We now prove that this definition is that same as the nicer non-computational one.
```

Comment:
s/that same/the same/

Reply:
Moved domineering to a separate PR, and fixed there.